### PR TITLE
add scratchpad indicator

### LINKED ISSUE #63: add an scratchpad-indicator to waybar
This is really just a cosmetic thing and rather a nice to have. There is a [reddit post](https://www.reddit.com/r/swaywm/comments/re89s3/waybar_scratchpad_indicator/) about an[ scratchpad-indicator.](https://github.com/justinesmithies/sway-dotfiles/commit/c8cba5215f0e94c51a6d6e3c9ad5fe373edc0ace#diff-cb4b857ece2e4549faa5ceb3370836f92b04487565f50cc9ce589d6c9e6d4810) I use it and it makes sense just to have a little visual reminder that one has an active scratchpad running.
Maybe you have some alternatives that I should also look at, but for now I am quite happy with this tool.

### PR DESCRIPTION
**Pull request by denisok**
fixes #63 

image

### CODE REVIEW

File: .config/waybar/scratchpad-indicator.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+output="$(swaymsg -t get_tree | jq -r 'recurse(.nodes[]?)|recurse(.floating_nodes[]?)|select(.name == "__i3_scratch").floating_nodes[]|"`<b>"+.name+"`</b> - "+.app_id+" "+(.id|tostring)')"
+number="$(swaymsg -r -t get_tree | jq -r 'recurse(.nodes[]) | first(select(.name=="__i3_scratch")) | .floating_nodes | length')"
+
+if [ "$number" -gt 0 ]; then
```
Comment (FilippoBonazziSUSE) [inline .config/waybar/scratchpad-indicator.sh:6]: The scratchpad indicator won't show up when $number==0 -> text =="", meaning that no right-click "move to scratchpad" action is possible when there are no windows in the scratchpad already

Comment (taschenlampe) [reply .config/waybar/scratchpad-indicator.sh:6]: Hi! @FilippoBonazziSUSE I am struggling to understand and to reproduce what you describe. Could you be so kind as to explain step by step what you do and where the script fails?

Comment (FilippoBonazziSUSE) [reply .config/waybar/scratchpad-indicator.sh:6]: Hi @taschenlampe, I was referring to the fact that the scratchpad indicator has a right-click action to move the active window to the scratchpad. However, when the scratchpad is empty, the scratchpad indicator won't show up (by design), so this right-click context action is not accessible.

The right-click context action is nice for user discoverability, but if it doesn't show up in the default state it's not discoverable :smile: .

I personally don't use the right-click action, so it doesn't bother me. In the end it's a trade-off between space efficiency (don't show the scratchpad indicator when empty) and a nice-to-have feature (right click indicator to move window to scratchpad).

Comment (taschenlampe) [reply .config/waybar/scratchpad-indicator.sh:6]: @FilippoBonazziSUSE  Oh I see now!
 My thought: "New" users may not know what a scratchpad is and will be confused of having something in the bar indicator that mostly will be unused. So I´d say, it's a nice little feature for the ones that know about scratchpad, they'll enjoy this little detail :-)

Comment (FilippoBonazziSUSE) [reply .config/waybar/scratchpad-indicator.sh:6]: Sure! Let's go ahead with it then

**Pull request merged**